### ISSUE TITLE
Mpris feature is not working

### ISSUE THREAD

**Issue by glaubersm** [celluloid-player/celluloid#452]
Overview Description:
KDE Plasma DE has an applet in systray called Media Player.
With it we can control the playback with any player that supports mpris
(see it working with VLC Player in the screenshot below).
However, Gnome MPV can not be controlled with such applet.

Steps to Reproduce:
1. install Gnome MPV on a distribution running KDE Plasma desktop environment
2. play any audio/video with Gnome MPV
3.

Actual Results:
we can not control Gnome MPV through the Media Player applet in systray.
Media Player applet does not detect Gnome MPV.
Expected Results:
Gnome MPV can be controlled on KDE Plasma through its Media Player applet

Version: 0.16

Additional Information:
I use Arch Linux, KDE Plasma 5.16.4 desktop environment.

![mpris](https://user-images.githubusercontent.com/15967631/62480137-6d12e900-b785-11e9-966e-72abcef688a0.jpg)

**Comment by gnome-mpv** [comment]
I tested this with other GTK-based players (Gnome Music and LollyPop) and they all failed to appear in the widget. Yet, they all work fine with Gnome Shell and [mpristester](https://github.com/randomguy3/mpristester). I also notice that Plasma logs an error message like this for all of them:
```
"org.mpris.MediaPlayer2.io.github.celluloid_player.Celluloid.instance-1" does not implement org.freedesktop.DBus.Properties correctly Error message was "org.freedesktop.DBus.Error.UnknownMethod" "No such interface “org.freedesktop.DBus.Properties” on object at path /org/mpris/MediaPlayer2"
```

So I think this is probably either a bug in GTK/GIO or KDE's MPRIS implementation.

**Comment by glaubersm** [comment]
humm, Lollypop and Gnome Music work on my system.

![lollypop](https://user-images.githubusercontent.com/15967631/62487597-a30d9880-b798-11e9-80a8-5c595e4c5f7b.png)

![gnomemusic](https://user-images.githubusercontent.com/15967631/62487600-a6a11f80-b798-11e9-8a83-fdc97b6550da.png)

**Comment by gnome-mpv** [comment]
I just tested again with another computer (using the exact same VM image) and surprisingly it kind of works this time. More specifically, Celluloid and Gnome Music works, but only if they are launched from the "start menu" rather than from command line. Lollypop still doesn't work.

I'm not sure what could be causing this sort of behavior, but I suspect there's some sort of race condition going on.

**Comment by glaubersm** [comment]
Hummm, on my system celluloid-git does not work with Media Player applet even when launched from the apps launcher.

I have another computer running Arch Linux + Gnome 3.32.2 + celluloid-git.
Playback controls are missing in notification area of Gnome.

**Comment by baybal** [comment]
Do you use mpv-mpris plugin by any chance?

**Comment by baybal** [comment]
Definitely something going on in gnome-mpv. @gnome-mpv what is your setup?
```
 baybal@xps  ~  playerctl -a metadata               
HUAWEI_MAIMANG_5 xesam:title               Encounter
HUAWEI_MAIMANG_5 xesam:trackNumber         2
HUAWEI_MAIMANG_5 mpris:length              139349000
HUAWEI_MAIMANG_5 xesam:album               Pump Audio
HUAWEI_MAIMANG_5 xesam:artist              The Cheebacabra - Pump Audio
HUAWEI_MAIMANG_5 xesam:genre               

(playerctl:13911): GLib-GIO-WARNING **: 21:48:31.812: Received property Position with type i does not match expected type x in the expected interface
io.github.GnomeMpv xesam:url                 
io.github.GnomeMpv mpris:length              0
io.github.GnomeMpv mpris:trackid             '/io/github/GnomeMpv/Track/0'
```

**Comment by glaubersm** [comment]
mpv-mpris plugin is not installed on my system.

**Comment by gnome-mpv** [comment]
@baybal: I'm not using the plugin.

**Comment by baybal** [comment]
I just compared mpv-mpris with gnome-mpv side by side. mpv-mpris gets properly recognised by xfce4-pulseaudio-plugin, and gets UI shown, but not gnome-mpv.

**Comment by baybal** [comment]
Any way I can help with debugging?

**Comment by gnome-mpv** [comment]
@baybal: One thing that might be helpful would be to compare the sequence of D-Bus messages sent/received by Celluloid and mpv-mpris (and maybe other mpris clients too) and see if there's anything that looks off.

**Comment by JJRcop** [comment]
I'm on KDE. `playerctl` works from the terminal, but KDE's media key shortcuts which link to MPRIS don't work, and the plasmoid applet listed in the original bug still doesn't work. (celluloid 0.20)

Tried other Gnome players: totem, lollypop, rhythmbox. Of the 3, lollypop worked.

**Comment by mymedia2** [comment]
This issue should be fixed in commit 92f3d8cc0a0e154a831a062a10b0167015002160.

**Comment by glaubersm** [comment]
Right. I'm unable to reproduce with Celluloid 0.23 on Arch Linux.
